pipenv/cli/command.py

```diff
@@ -569,7 +569,10 @@ def run_open(state, module, *args, **kwargs):
     else:
         p = c.out.strip().rstrip("cdo")
     echo(crayons.normal("Opening {0!r} in your EDITOR.".format(p), bold=True))
-    edit(filename=p)
+    from ..core import inline_activate_virtual_environment
+    inline_activate_virtual_environment()
+    environment={"VIRTUAL_ENV":os.environ["VIRTUAL_ENV"]}
```

Comment:
Is this line required? I would expect Click to automatically forward `os.environ` to the editor process (it sounds like a bug if it doesn’t).

Reply:
Also, should this also pass `.env` to the editor? (I am not entirely sure on this one, just throwing out the idea.)

Reply:
As far as I can tell `run_open` uses the `edit` implementation from click's `termui`.
https://github.com/pypa/pipenv/blob/55856c70ace5caf7c1b53d253fbc7c4fa7421ac0/pipenv/vendor/click/termui.py#L481

This function has several optional arguments including the `env` argument which is where the user passes in environment variables they want to pass into the editor. By default all the arguments are set to `None` so by default the environment variables are not passed in to the `Editor` object.

I can confirm that Click does not automatically forward `os.environ`. 

You can find the code that handles the environment variables [here](https://github.com/pallets/click/blob/b9a72b4b234b4a8090c7ef085375658ee12aaffe/click/_termui_impl.py#L421-L425) on lines 421-425 of `_termui_impl.py`. `os.environ` and it's contents is only passed to the editor if the `Editor`'s `env` variable is defined in which case it copies the contents of `os.environ` appends it's own `env` variables and passes that into the editor. If it's not the case it passes `None` to the editor's environment. 

This does sound like a bug on Click's side. Should we close this PR and I'll create a PR to Click fixing the bug?

Reply:
Hmm. The implementation of `_termui_impl` calls `subprocess.Popen` with `env`. According to the docs, `Popen` defaults to “inheriting the current process’ environment” if `env` is None, so I think Click is doing what it intends to. Can you try skip setting `env`, and see if calling `inline_activate_virtual_environment()` is enough to initialise the subprocess?